packages/api-server/src/__tests__/fastify.test.ts

```diff
@@ -0,0 +1,99 @@
+import fastify from 'fastify'
+import { vol } from 'memfs'
+
+import { createFastifyInstance, DEFAULT_OPTIONS } from '../fastify'
+
+// Suppress terminal logging.
+console.log = jest.fn()
+
+// Set up RWJS_CWD.
+let original_RWJS_CWD
+const redwoodProjectPath = '/redwood-app'
+
+beforeAll(() => {
+  original_RWJS_CWD = process.env.RWJS_CWD
+
+  process.env.RWJS_CWD = redwoodProjectPath
+})
+
+afterAll(() => {
+  process.env.RWJS_CWD = original_RWJS_CWD
+})
+
+// Mock server.config.js.
+jest.mock('fs', () => require('memfs').fs)
+
+const userConfig = {
+  requestTimeout: 25_000,
+}
+
+jest.mock(
+  '/redwood-app/api/server.config.js',
+  () => {
+    return {
+      config: userConfig,
+    }
+  },
+  {
+    virtual: true,
+  }
+)
+
+jest.mock(
+  '\\redwood-app\\api\\server.config.js',
+  () => {
+    return {
+      config: userConfig,
+    }
+  },
+  {
+    virtual: true,
+  }
+)
+
+jest.mock('fastify', () => {
+  return jest.fn(() => {
+    return {
+      register: () => {},
+    }
+  })
+})
+
+describe('createFastifyInstance', () => {
+  it('instantiates a fastify instance with default config', () => {
+    vol.fromNestedJSON(
+      {
+        'redwood.toml': '',
+      },
+      redwoodProjectPath
+    )
+
+    createFastifyInstance()
+    expect(fastify).toHaveBeenCalledWith(DEFAULT_OPTIONS)
+  })
+
+  it("instantiates a fastify instance with the user's configuration if available", () => {
+    vol.fromNestedJSON(
+      {
+        'redwood.toml': '',
+        api: {
+          'server.config.js': '',
+        },
+      },
+      redwoodProjectPath
+    )
+
+    createFastifyInstance()
+    expect(fastify).toHaveBeenCalledWith(userConfig)
+  })
+})
+
+test('DEFAULT_OPTIONS configures the log level based on NODE_ENV', () => {
```

Comment:
This test doesn't strictly test that the returned value changes based on the value of `NODE_ENV` but I can imagine it being a little more difficult to test this if the module is cached across test cases.

Reply:
[Jest sets `NODE_ENV` to `'test'`](https://jestjs.io/docs/environment-variables#node_env), so it does test the logic indirectly. But yeah I didn't want to change `NODE_ENV`.